### PR TITLE
core/service: make service_add_fd_store() always consume provided fd

### PR DESCRIPTION
**Pull request by yuwata**
No functional change, just refactoring.

### CODE REVIEW

File: src/core/service.c [hunk 1]
```diff
@@ -480,12 +481,13 @@ static int on_fd_store_io(sd_event_source *e, int fd, uint32_t revents, void *us
         return 0;
 }
 
-static int service_add_fd_store(Service *s, int fd, const char *name, bool do_poll) {
+static int service_add_fd_store(Service *s, int fd_in, const char *name, bool do_poll) {
+        _cleanup_(service_fd_store_unlinkp) ServiceFDStore *fs = NULL;
+        _cleanup_(asynchronous_closep) int fd = fd_in;
```
Comment (YHNdnzj) [inline src/core/service.c]: Maybe use `ASSERT_FD` here?

File: src/core/service.c [hunk 2]
```diff
@@ -480,12 +481,13 @@ static int on_fd_store_io(sd_event_source *e, int fd, uint32_t revents, void *us
...
         struct stat st;
-        ServiceFDStore *fs;
         int r;
 
-        /* fd is always consumed if we return >= 0 */
+        /* fd is always consumed even if failed. */
 
         assert(s);
         assert(fd >= 0);
```
Comment (YHNdnzj) [inline src/core/service.c]: (And remove this one)

**Review by YHNdnzj — APPROVED** [review]

**Comment by YHNdnzj** [comment]
```
systemd[1]: systemd-journald.service: Got notification message from PID 708 (FDSTORE=1)
systemd[1]: systemd-journald.service: Trying to stash fd for dev=0:8/inode=17707
systemd[1]: Assertion 'fd >= 0' failed at src/libsystemd/sd-event/sd-event.c:1249, function sd_event_add_io(). Ignoring.
systemd[1]: systemd-journald.service: Failed to add fd to store: Bad file descriptor
```

**Comment by yuwata** [comment]
@YHNdnzj Thank you for your review. The suggested changes are applied, and fixed test failure.

**Comment by poettering** [comment]
lgtm

**Comment by keszybz** [comment]
TEST-36-NUMAPOLICY:                 FAIL     (1804 s)
TEST-50-DISSECT:                    FAIL     (1813 s)
Looks unrelated.

File: src/core/service.c [hunk 3]
```diff
@@ -480,15 +481,15 @@ static int on_fd_store_io(sd_event_source *e, int fd, uint32_t revents, void *us
...
-static int service_add_fd_store(Service *s, int fd, const char *name, bool do_poll) {
+static int service_add_fd_store(Service *s, int fd_in, const char *name, bool do_poll) {
+        _cleanup_(service_fd_store_unlinkp) ServiceFDStore *fs = NULL;
+        _cleanup_(asynchronous_closep) int fd = ASSERT_FD(fd_in);
         struct stat st;
-        ServiceFDStore *fs;
         int r;
 
-        /* fd is always consumed if we return >= 0 */
+        /* fd is always consumed even if failed. */
```
Comment (keszybz) [inline src/core/service.c]: "if failed" would mean that the fd is failed.
"if the function fails" or "if we fail" or "if this fails" or "on failure".

**Pull request merged**